### DISCUSSION TITLE
SPDM specification 1.3 planning for libspdm

### DISCUSSION THREAD

**Discussion by jyao1** [DMTF/libspdm#2276] [Announcements]
libspdm starts planning for SPDM specification 1.3.

1) Add SPDM 1.3 support for existing command.
 - Add version 1.3 version negotiation
 - New data fields for each command (CHANLENGE, MEASUREMENTS, CSR, SET_CERT, etc)
 - PSK session key derivation update (use all-0xFF for salt_0)
 - Transcript and key schedule version text update to 1.3

2) Add SPDM 1.3 support for new feature
 - Multiple Key Support (`MULTI_KEY_CAP`)
 - Generic Certificate Mode
 - Key Configuration (`GET_KEY_PAIR_INFO_CAP`, `SET_KEY_PAIR_INFO_CAP`)
 - Event support (`EVENT_CAP`)
 - Endpoint Info (`EP_INFO_CAP`)
 - Hash-Extend Measurement (HEM)
 - Measurement Extension Log (MEL) (`MEL_CAP`)
 
Any contribution is welcome!

Issue List:
* https://github.com/DMTF/libspdm/issues/2279
* https://github.com/DMTF/libspdm/issues/2280
* https://github.com/DMTF/libspdm/issues/2281
* https://github.com/DMTF/libspdm/issues/2282
* https://github.com/DMTF/libspdm/issues/2283
* https://github.com/DMTF/libspdm/issues/2284
* https://github.com/DMTF/libspdm/issues/2285
* https://github.com/DMTF/libspdm/issues/2286
* https://github.com/DMTF/libspdm/issues/2287
* https://github.com/DMTF/libspdm/issues/2288
* https://github.com/DMTF/libspdm/issues/2289
* https://github.com/DMTF/libspdm/issues/2290
* https://github.com/DMTF/libspdm/issues/2291
* https://github.com/DMTF/libspdm/issues/2292
* https://github.com/DMTF/libspdm/issues/2293
* https://github.com/DMTF/libspdm/issues/2294
* https://github.com/DMTF/libspdm/issues/2295
* https://github.com/DMTF/libspdm/issues/2296

**Comment by steven-bellock** [comment]
For eventing, multikey, and MEL / HEM, let's spawn separate design discussions. I'm a glutton for punishment so I'll pick up eventing.
